docs/contributing/999-rfcs/2024-03-31-wing-secrets-cli.md

```diff
@@ -0,0 +1,116 @@
+---
+title: "#6105 Wing Secrets CLI"
+description: Creating Wing secrets from the CLI
+---
+
+# Wing Secrets CLI
+- **Author(s)**: @hasanaburayyan
+- **Submission Date**: 2024-03-31
+- **Stage**: Draft
+
+Creating secrets through the Wing CLI.
+
+## Background
+
+Wing applications often require secrets to be retrieved during runtime. These secrets are stored in platform specific secret stores, such as AWS Secrets Manager for `tf-aws` or a local `.env` file for the `sim` platform.
+
+Secrets must be configured before the application is run, and now the Wing CLI along with Wing platforms make it easy to configure secrets.
+
+### Out of Scope
+
+In this RFC a few things are out of scope:
+- Checking if the secrets exist in the platform's secret store when running `wing compile`
+- Reading secret values, for now we will only focus on creating secrets
+
+## Platform Hook
+
+Since secrets creation is platform specific, platforms can now implement a new hook `configureSecrets(secrets: { [key: string]: string }): string` which will be called by the Wing CLI to configure the secrets.
+
+For example the `sim` platform implementation which needs to store secrets in a `.env` file, would look something like this:
+
+```js
+public async createSecrets(secrets: { [key: string]: string }): Promise<string> {
+  let existingSecretsContent = "";
+  try {
+    existingSecretsContent = fs.readFileSync('./.env', 'utf8');
+  } catch (error) {}
+
+  const existingSecrets = existingSecretsContent.split('\n')
+    .filter(line => line.trim() !== '')
+    .reduce((s, line) => {
+      const [key, value] = line.split('=', 2);
+      s[key] = value;
+      return s;
+    }, {} as { [key: string]: string });
+
+  for (const key in secrets) {
+    existingSecrets[key] = secrets[key];
+  }
+
+  const updatedContent = Object.entries(existingSecrets)
+    .map(([key, value]) => `${key}=${value}`)
+    .join('\n');
+
+  fs.writeFileSync('./.env', updatedContent);
+
+  return "Secrets saved to .env file";
```

Comment:
I think a simple `console.log` here should be fine. What's the motivation of returning the output?

Reply:
I agree that a `console.log` is probably fine, my initial motivation during the POC was to keep all the logging output and formatting within the cli command. I also figured requiring a return string might serve to motivate platform authors to provide useful response messages.

Im not married to the idea of the string return, happy to re-evaluate in the implementation PR